packages/backend/src/webhook/service.ts

```diff
@@ -0,0 +1,225 @@
+  Object.values(PaymentEventType).includes(type)
+
+interface InvoiceEvent {
+  type: InvoiceEventType
+  invoice: Invoice
+  payment?: never
+  amountReceived: bigint
+  amountSent?: never
+}
+
+interface PaymentEvent {
+  type: PaymentEventType
+  invoice?: never
+  payment: OutgoingPayment
+  amountReceived?: never
+  amountSent: bigint
+}
+
+export type EventOptions = InvoiceEvent | PaymentEvent
+
+interface InvoiceData {
+  invoice: {
+    id: string
+    accountId: string
+    active: boolean
+    description?: string
+    createdAt: string
+    expiresAt: string
+    amount: string
+    received: string
+  }
+  payment?: never
+}
+
+interface PaymentData {
+  invoice?: never
+  payment: {
+    id: string
+    accountId: string
+    createdAt: string
+    state: PaymentState
+    error?: string
+    stateAttempts: number
+    intent: {
+      paymentPointer?: string
+      invoiceUrl?: string
+      amountToSend?: string
+      autoApprove: boolean
+    }
+
+    quote?: {
+      timestamp: string
+      activationDeadline: string
+      targetType: PaymentType
+      minDeliveryAmount: string
+      maxSourceAmount: string
+      maxPacketAmount: string
+      minExchangeRate: number
+      lowExchangeRateEstimate: number
+      highExchangeRateEstimate: number
+    }
+    destinationAccount: {
+      scale: number
+      code: string
+      url?: string
+    }
+    outcome: {
+      amountSent: string
+    }
+  }
+}
+
+interface WebhookEvent {
+  id: string
+  type: EventType
+  data: InvoiceData | PaymentData
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+export const isPaymentEvent = (event: any): event is PaymentEvent =>
+  Object.values(PaymentEventType).includes(event.type)
+
+export interface WebhookService {
+  send(options: EventOptions): Promise<void>
+}
+
+interface ServiceDependencies {
+  config: IAppConfig
+  logger: Logger
+}
+
+export async function createWebhookService(
+  deps_: ServiceDependencies
+): Promise<WebhookService> {
+  const logger = deps_.logger.child({
+    service: 'WebhookService'
+  })
+  const deps = { ...deps_, logger }
+  return {
+    send: (options) => sendWebhook(deps, options)
+  }
+}
+
+async function sendWebhook(
+  deps: ServiceDependencies,
+  options: EventOptions
+): Promise<void> {
+  const event = {
+    id: uuid(),
+    type: options.type,
+    data: isPaymentEvent(options)
+      ? paymentToData(options.payment, options.amountSent)
+      : invoiceToData(options.invoice, options.amountReceived)
+  }
+
+  const requestHeaders = {
+    'Content-Type': 'application/json'
+  }
+
+  if (deps.config.webhookSecret) {
+    requestHeaders['Rafiki-Signature'] = generateWebhookSignature(
+      event,
+      deps.config.webhookSecret,
+      deps.config.signatureVersion
+    )
+  }
+
+  await axios.post(deps.config.webhookUrl, event, {
+    headers: requestHeaders
```

Comment:
There should be a timeout on the outgoing request, I think axios defaults to no timeout.